qiskit/circuit/parameterexpression.py

```diff
@@ -303,49 +322,74 @@ def _call(self, ufunc):
 
     def sin(self):
         """Sine of a ParameterExpression"""
-        from sympy import sin as _sin
-        return self._call(_sin)
+        if HAS_SYMENGINE:
+            return self._call(symengine.sin)
+        else:
+            from sympy import sin as _sin
+            return self._call(_sin)
 
     def cos(self):
         """Cosine of a ParameterExpression"""
-        from sympy import cos as _cos
-        return self._call(_cos)
+        if HAS_SYMENGINE:
+            return self._call(symengine.cos)
+        else:
+            from sympy import cos as _cos
+            return self._call(_cos)
 
     def tan(self):
         """Tangent of a ParameterExpression"""
-        from sympy import tan as _tan
-        return self._call(_tan)
+        if HAS_SYMENGINE:
+            return self._call(symengine.tan)
+        else:
+            from sympy import tan as _tan
+            return self._call(_tan)
 
     def arcsin(self):
         """Arcsin of a ParameterExpression"""
-        from sympy import asin as _asin
-        return self._call(_asin)
+        if HAS_SYMENGINE:
+            return self._call(symengine.asin)
+        else:
+            from sympy import asin as _asin
+            return self._call(_asin)
 
     def arccos(self):
         """Arccos of a ParameterExpression"""
-        from sympy import acos as _acos
-        return self._call(_acos)
+        if HAS_SYMENGINE:
+            return self._call(symengine.acos)
+        else:
+            from sympy import acos as _acos
+            return self._call(_acos)
 
     def arctan(self):
         """Arctan of a ParameterExpression"""
-        from sympy import atan as _atan
-        return self._call(_atan)
+        if HAS_SYMENGINE:
+            return self._call(symengine.atan)
+        else:
+            from sympy import atan as _atan
+            return self._call(_atan)
 
     def exp(self):
         """Exponential of a ParameterExpression"""
-        from sympy import exp as _exp
-        return self._call(_exp)
+        if HAS_SYMENGINE:
+            return self._call(symengine.exp)
+        else:
+            from sympy import exp as _exp
+            return self._call(_exp)
 
     def log(self):
         """Logarithm of a ParameterExpression"""
-        from sympy import log as _log
-        return self._call(_log)
+        if HAS_SYMENGINE:
+            return self._call(symengine.log)
+        else:
+            from sympy import log as _log
+            return self._call(_log)
 
     def __repr__(self):
         return '{}({})'.format(self.__class__.__name__, str(self))
 
     def __str__(self):
-        return str(self._symbol_expr)
+        from sympy import sympify
```

Comment:
does symengine not have a string representation yet?

Reply:
It does, but it produces slightly different results which were breaking tests (and would cause consistency between systems using symengine and those not). For example, one test was failing with `phi + 2 != 2 + phi` when looking at the string representation. So I opted to just use a sympy string representation everywhere for consistency.

Reply:
ok